### PR TITLE
Update: option to ignore top-level max statements (fixes #4309)

### LINKED ISSUE #4309: `max-statements` should have option to ignore top-level function scope
The UMD/IIFE wrapper is a common pattern, but it's inconvenient to have to disable `max-statements` for the first line of the wrapper each time as a workaround for this problem. Example:

```js
/* eslint max-statements: [2, 20] */
(function (global) { // eslint-disable-line max-statements
  // code with most likely too many statements...
})(typeof window !== 'undefined' ? window : this);
```

There should be an option to ignore the top-level function scope, e.g. the scope that second comment is in, so the IIFE and UMD patterns can be better accommodated. Maybe like `"ignoreTopLevelFunctions": true` or something to that effect?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27869613-max-statements-should-have-option-to-ignore-top-level-function-scope?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: docs/rules/max-statements.md [hunk 1]
```diff
@@ -14,6 +14,12 @@ function foo() {
 
 This rule allows you to configure the maximum number of statements allowed in a function.  The default is 10.
 
+There is an additional optional argument to ignore top level functions.
```
Comment (nzakas) [inline docs/rules/max-statements.md]: Can you add an Options header here?

**Comment by michaelficarra** [comment]
The request here was just for an exception when the one and only statement at the top level is an IIFE. Having this affect each top level function is not really desirable.

**Comment by nzakas** [comment]
Good catch. Yeah, this should only apply if there's one top-level function.

**Comment by alberto** [comment]
@nzakas And only if it is an IIFE?

**Comment by nzakas** [comment]
No, any function (see AMD example in the issue).

**Comment by alberto** [comment]
I'm a bit lost, isn't the example an IIFE?

**Comment by platinumazure** [comment]
No, AMD does not necessary use IIFEs. Here's an example:

```js
define(["depencency1", "dependency2"], function (dependency1, dependency2) {
    // Five billion statements here, but each function declaration/expression within this function
    // should have the rule applied

    function someFunction() {
        // This function should have the rule applied
    }
});

// One statement total at top level

Basically, the `define()` function call above is exactly analogous to the typical IIFE case, for the purpose of the proposed extension to this rule. Does that make sense?

**Comment by alberto** [comment]
Thanks for clearing it up for me, @platinumazure :)

**Comment by nzakas** [comment]
Lgtm
